sdk/samples/iot/paho_iot_provisioning_sample.c

```diff
@@ -358,13 +346,28 @@ static void handle_device_registration_status_message(
     }
     else // Unsuccessful assignment (unassigned, failed or disabled states)
     {
+      uint8_t buffer[100];
+      az_span remainder;
+      az_span operation_status_span = AZ_SPAN_FROM_BUFFER(buffer);
+      if (az_result_failed(az_span_i32toa(
+              operation_status_span, register_response->operation_status, &remainder)))
+      {
+        operation_status_span = AZ_SPAN_FROM_STR("UNKNOWN");
+      }
+      else
+      {
+        remainder = az_span_copy_u8(remainder, 0);
+        operation_status_span = az_span_slice(
+            operation_status_span,
+            0,
+            az_span_size(operation_status_span) - az_span_size(remainder));
+      }
       IOT_SAMPLE_LOG_ERROR("Device provisioning failed:");
-      IOT_SAMPLE_LOG_AZ_SPAN("Registration state:", register_response->operation_status);
+      IOT_SAMPLE_LOG_AZ_SPAN("Registration state:", operation_status_span);
```

Comment:
Do we need to first turn the enum into a span and then log it as a span? Why not log the integer directly?

Reply:
We can log the integer here directly since it's an enum.    Will use an IOT_SAMPLE_LOG() instead of IOT_SAMPLE_LOG_AZ_SPAN().
Another option is to create an internal API function that gets the operation status as an az_span ("unassigned", "failed", etc) given an enum.  A print out of an integer doesn't tell the user much.  They have to go look up what the enum means.

Reply:
I'd say just log the enum directly. The enums would start from 0 and the user would be able to just look up in the typedef pretty easily. Most debuggers also will tell you the enum text if you set a breakpoint. It's not worth the overhead of the conversion.